core/src/banking_stage/forwarder.rs

```diff
@@ -167,32 +167,18 @@ impl Forwarder {
         if !packet_vec.is_empty() {
             inc_new_counter_info!("banking_stage-forwarded_packets", packet_vec_len);
 
-            let mut measure = Measure::start("banking_stage-forward-us");
-
-            let res = if let ForwardOption::ForwardTpuVote = forward_option {
-                // The vote must be forwarded using only UDP.
+            let (res, forward_us) = measure_us!(self.forward(forward_option, packet_vec, &addr));
+            if let ForwardOption::ForwardTpuVote = forward_option {
                 banking_stage_stats
```

Comment:
I think it makes sense to move the stats update logic banking_stage_stats into the new forward function as well to make it more self contained.

Reply:
Separating the stats update from the forwarding was actually the point. New worker threads won't use `BankingStageStats` struct.

https://github.com/solana-labs/solana/pull/30925 further abstracts so we have a fn that does leader check, data budget update, filtering, forwarding and returns just the relavent metrics (successful?, how many packets, time, etc) - intent is to use that fn from the worker threads.